npm_and_yarn/lib/dependabot/npm_and_yarn/version_selector.rb

```diff
@@ -0,0 +1,45 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "dependabot/shared_helpers"
+
+module Dependabot
+  module NpmAndYarn
+    class VersionSelector
+      extend T::Sig
+      extend T::Helpers
+
+      # For limited testing, allowing only specific versions defined in engines in package.json
+      # such as "20.8.7", "8.1.2", "8.21.2",
+      NODE_ENGINE_SUPPORTED_REGEX = /^\d+(?:\.\d+)*$/
+
+      sig { params(manifest_json: T.untyped, name: String).returns(T::Hash[Symbol, T.untyped]) }
```

Comment:
I think you can be more specific here

```suggestion
      sig { params(manifest_json: T::Hash[String, String], name: String).returns(T::Hash[Symbol, T.untyped]) }
```